app/controllers/resources_controller.rb

```diff
@@ -36,6 +36,21 @@ def push_to_onesky
 
   private
 
+  def cached_index_json
+    cache_key = Resource.index_cache_key(all_resources,
+      include_param: params[:include],
+      fields_param: field_params)
+    Rails.cache.fetch(cache_key, expires_in: 1.hour) { index_json }
+  end
+
+  def index_json
+    ActiveModelSerializers::SerializableResource.new(
+      all_resources.order(name: :asc),
+      include: params[:include],
+      fields: field_params
+    ).to_json
+  end
+
   def all_resources
     if params["filter"]
```

Comment:
`filter` params are not currently being observed in the `cache_key`, right now this doesn't matter too much because everything is in the same `system` that we would filter on. But we have had discussions around leveraging `system` to provide different collections of tools, which would mean similar requests with different filter params.

Reply:
it actually subtly is. because I am sending the result of this method into the cache key calculation and calling `cache_key_with_version` on it. Under the hood eventually rails calls `to_sql` on the relation and hashed the resulting sql for the cache key: https://github.com/rails/rails/blob/v6.0.3.7/activerecord/lib/active_record/relation.rb#L317

Reply:
nice, didn't realize this (even though you mentioned it in your PR description...)

Reply:
maybe you missed it, or maybe you just didn't believe me ;)

Reply:
It's often difficult to believe @soberstadt, but most of the time, he is correct. ;)